### PR TITLE
Bug 1259681 - Enable spatial navigation for custom elements in fxos-t…

### PR DESCRIPTION
**Pull request by begeeben**
Bug 1259681 - Enable spatial navigation for custom elements in fxos-tv-modal-dialog, r=luke

### CODE REVIEW

File: fxos-tv-input-dialog.js [hunk 1]
```diff
@@ -41,13 +39,25 @@
...
     FxosTvModalDialog.prototype._render.call(this);
     this.textInput = document.createElement('input');
-    this.textInput.addEventListener('keydown', function(e) {
-      if (e.keyCode === KeyEvent.DOM_VK_RIGHT) {
-        if (this.selectionEnd < this.value.length) {
+    this.textInput.addEventListener('keydown', (e) => {
+      switch (e.keyCode) {
+        case KeyEvent.DOM_VK_RIGHT:
           // We are going to move cursor in the text input, calling
           // stopPropagation to prevent focus being transfer to clear button.
```
Comment (luke-chang) [inline fxos-tv-input-dialog.js]: nit: ... prevent focus _from_ being _transferred_ ...

File: fxos-tv-input-dialog.js [hunk 2]
```diff
@@ -41,13 +39,25 @@
...
+          if (this.textInput.selectionEnd < this.textInput.value.length) {
+            e.stopPropagation();
+          }
+          break;
+        case KeyEvent.DOM_VK_LEFT:
+          // We are going to move cursor in the text input, calling
+          // stopPropagation to prevent focus being transfer out of the input.
```
Comment (luke-chang) [inline fxos-tv-input-dialog.js]: same as above.

File: fxos-tv-input-dialog.js [hunk 3]
```diff
@@ -154,15 +161,18 @@
...
+        }
       }.bind(this));
       this.buttonElements.push(button);
       this.buttonGroup.appendChild(button);
 
       var customFocus = this.customSettings && this.customSettings.defaultFocus;
-      if (buttonSetting.defaultFocus && !customFocus) {
-        this._focusedIndex[BUTTON_GROUP_INDEX] = index;
+      if ((!this.defaultFocusElement &&
+          !customFocus) ||
```
Comment (luke-chang) [inline fxos-tv-input-dialog.js]: unnecessary line break?

File: fxos-tv-modal-dialog.js [hunk 4]
```diff
@@ -132,20 +143,23 @@
...
-        this._focusedIndex = 0;
-      }
-      this.focus();
-    }
+    var id = '#' + this.element.id;
+    SpatialNavigation.init();
+    SpatialNavigation.add({
+      id: this.element.id,
+      selector: id + ' button, ' + id + ' input, ' + id + ' fxos-tv-button',
+      // enterTo: 'last-focused',
```
Comment (luke-chang) [inline fxos-tv-modal-dialog.js]: remove comments.

File: fxos-tv-modal-dialog.js [hunk 5]
```diff
@@ -132,20 +143,23 @@
...
+      restrict: 'self-only'
+    });
+
+    SpatialNavigation.makeFocusable();
+    SpatialNavigation.focus(this.defaultFocusElement);
```
Comment (luke-chang) [inline fxos-tv-modal-dialog.js:161]: The first navigable element will be focused if `this.defaultFocusElement` is `null`. I just want to confirm if that's what you expect.

Comment (begeeben) [inline fxos-tv-modal-dialog.js:161]: Thanks! It will only be invoked after the default focus is ready.

**Pull request merged**